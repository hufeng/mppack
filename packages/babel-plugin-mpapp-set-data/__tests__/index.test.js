const babel = require('babel-core');
const setData = require('../index');

const testCode = `
//test mixin
const TabMixin = Mixin({
  data: {
    tabIndex: 0
  },

  onHandleTabIndex(tabIndex) {
   this.setData({
     tabIndex
   });
   this.$spliceData({})
  }
});

//test plain javascript object
const pojo = {
  data: {
    test: 0
  },

  sayHello() {
    this.setData({})
    this.$spliceData({})
  }
};

//test mue object
Mue({
  data: {
    list: []
  },

  sayHello() {
    this.setData({});
    this.$spliceData({})
  }
});

Action({
  sayHello() {
    this.setData({});
    this.$spliceData({})
  }
})
`;

it('test setData and spliceData', () => {
  const { code } = babel.transform(testCode, {
    plugins: [setData]
  });
  expect(code).toMatchSnapshot();
});
