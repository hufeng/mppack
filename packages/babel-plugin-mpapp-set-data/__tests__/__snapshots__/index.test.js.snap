// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test setData and spliceData 1`] = `
"
//test mixin
const TabMixin = Mixin({
  data: {
    tabIndex: 0
  },

  onHandleTabIndex(tabIndex) {
    this.setState({
      tabIndex
    });
    this.spliceState({});
  }
});

//test plain javascript object
const pojo = {
  data: {
    test: 0
  },

  sayHello() {
    this.setData({});
    this.$spliceData({});
  }
};

//test mue object
Mue({
  data: {
    list: []
  },

  sayHello() {
    this.setState({});
    this.spliceState({});
  }
});

Action({
  sayHello() {
    this.setState({});
    this.spliceState({});
  }
});"
`;
